### PR TITLE
[DETR, YOLOS] Fix device bug

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR fixes a device bug for DETR and YOLOS's `post_process_object_detection` methods, which currently give a device mismatch between CPU and CUDA when running the model on CUDA.

The PR also makes sure the postprocess methods are tested in the integration tests.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for adding these tests and fix for device! 

Only question is whether the integration test outputs have been sense checked.

File: tests/models/yolos/test_modeling_yolos.py [hunk 1]
```diff
@@ -373,3 +373,16 @@ def test_inference_object_detection_head(self):
         )
         self.assertTrue(torch.allclose(outputs.logits[0, :3, :3], expected_slice_logits, atol=1e-4))
         self.assertTrue(torch.allclose(outputs.pred_boxes[0, :3, :3], expected_slice_boxes, atol=1e-4))
+
+        # verify postprocessing
+        results = feature_extractor.post_process_object_detection(
```
[the remark this answers is not in the document]
Comment (NielsRogge) [reply tests/models/yolos/test_modeling_yolos.py:378]: There are still 272 files containing "feature_extractor": https://github.com/search?q=repo%3Ahuggingface%2Ftransformers%20feature_extractor&type=code

[the remark this answers is not in the document]
Comment (amyeroberts) [reply tests/models/yolos/test_modeling_yolos.py:378]: Yep - but quite a few of them are for audio / warnings  about them being deprecated :) I did a sweep of the ones in tests, when I renamed the test files but I obviously still missed some

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Review by alaradirik — APPROVED** [review]
Thanks for fixing it!

File: tests/models/detr/test_modeling_detr.py [hunk 2]
```diff
@@ -553,6 +554,19 @@ def test_inference_object_detection_head(self):
         ).to(torch_device)
         self.assertTrue(torch.allclose(outputs.pred_boxes[0, :3, :3], expected_slice_boxes, atol=1e-4))
 
+        # verify postprocessing
+        results = feature_extractor.post_process_object_detection(
+            outputs, threshold=0.3, target_sizes=[image.size[::-1]]
+        )[0]
+        expected_scores = torch.tensor([0.9982, 0.9960, 0.9955, 0.9988, 0.9987]).to(torch_device)
+        expected_labels = [75, 75, 63, 17, 17]
```
[the remark this answers is not in the document]
Comment (NielsRogge) [reply tests/models/detr/test_modeling_detr.py:562]: Yes :)

**Pull request merged**